perfkitbenchmarker/linux_benchmarks/aws_dynamodb_ycsb_benchmark.py

```diff
@@ -0,0 +1,161 @@
+AWS_CREDENTIAL_DIR = '/tmp/AWSCredentials.properties'
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('aws_dynamodb_ycsb_awscredentials_properties',
+                    './AWSCredentials.properties',
+                    'The AWS credential location.')
+flags.DEFINE_string('aws_dynamodb_ycsb_dynamodb_primarykey',
+                    'user01',
+                    'The primaryKey of table.')
+flags.DEFINE_string('aws_dynamodb_ycsb_dynamodb_endpoint',
+                    None,
+                    'The aws dynamodb endpoint to connect to.')
+flags.DEFINE_string('aws_dynamodb_ycsb_requestdistribution',
+                    'uniform',
+                    'Type of request distribution.')
+flags.DEFINE_string('aws_dynamodb_ycsb_readproportion',
+                    '0.5',
+                    'The Read Proportion, default is 0.5 in workloada and 0.95 in YCSB.')
+flags.DEFINE_string('aws_dynamodb_ycsb_updateproportion',
+                    '0.5',
+                    'The Update Proportion, default is 0.5 in workloada and 0.05 in YCSB')
+flags.DEFINE_string('aws_dynamodb_ycsb_table',
+                    'usertable',
+                    'The Table flag determines which table to run on.')
+flags.DEFINE_string('aws_dynamodb_ycsb_consistentReads',
+                    'false',
+                    'Consistent reads cost 2x Eventual reads. False is default and eventual')
+flags.DEFINE_string('aws_dynamodb_ycsb_capacity',
+                    'ReadCapacityUnits=5,WriteCapacityUnits=5',
+                    'Set RCU/WCU for table')
+
+
+def GetConfig(user_config):
+    config = configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+    if FLAGS['ycsb_client_vms'].present:
+        config['vm_groups']['default']['vm_count'] = FLAGS.ycsb_client_vms
+    return config
+
+
+def CheckPrerequisites(benchmark_config):
+    """Verifies that the required resources are present.
+    Raises:
+    perfkitbenchmarker.data.ResourceNotFound: On missing resource.
+    """
+    if not FLAGS.aws_dynamodb_ycsb_awscredentials_properties:
+        raise ValueError('"aws_dynamodb_ycsb_awscredentials_properties" must be set')
+    if not FLAGS.aws_dynamodb_ycsb_dynamodb_primarykey:
+        raise ValueError('"aws_dynamodb_ycsb_dynamodb_primarykey" must be set')
+    if not FLAGS.aws_dynamodb_ycsb_dynamodb_endpoint:
+        raise ValueError('"aws_dynamodb_ycsb_dynamodb_endpoint" must be set ')
+
+
+def Prepare(benchmark_spec):
+    """Install YCSB on the target vm.
+    Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+    """
+    benchmark_spec.always_call_cleanup = True
+    benchmark_spec.dynamodb_instance = aws_dynamodb.AwsDynamoDBInstance(
+        table_name=FLAGS.aws_dynamodb_ycsb_table,
+        primary_key=FLAGS.aws_dynamodb_ycsb_dynamodb_primarykey,
+        throughput=FLAGS.aws_dynamodb_ycsb_capacity)
+#    if benchmark_spec.dynamodb_instance._Exists():
```

Comment:
Cleanup unused code blocks ? Is this pull request ready?

Reply:
cleaned up